manim/utils/bezier.py

```diff
@@ -369,3 +372,114 @@ def get_quadratic_approximation_of_cubic(a0, h0, h1, a1):
 
 def is_closed(points: typing.Tuple[np.ndarray, np.ndarray]) -> bool:
     return np.allclose(points[0], points[-1])
+
+
+def bez_params_from_point(
+    point: typing.Iterable[typing.Union[float, int]],
+    control_points: typing.Iterable[typing.Iterable[typing.Union[float, int]]],
+) -> list[float]:
+    """Obtains the parameter corresponding to a given point
+    for a given bezier curve.
+
+    Parameters
+    ----------
+    point (typing.Iterable[float])
+        The Cartesian Coordinates of the point whose parameter
+        should be obtained.
+
+    control_points (typing.Iterable[typing.Iterable[float]])
+        The Cartesian Coordinates of the ordered control
+        points of the bezier curve on which the point may
+        or may not lie.
+
+    Returns
+    -------
+        list[float]
+            List containing possible parameters for the given point on
+            the given bezier curve.
+            This usually only contains one or zero elements, but if the
+            point is, say, at the beginning/end of a closed loop, may return
+            a list with more than 1 value, corresponding to the beginning and
+            end etc. of the loop.
+    """
+    assert all(np.shape(point) == np.shape(c_p) for c_p in control_points)
+
+    roots = []
+    for i in range(len(point)):
+        p = point[i]
+        # For now, we only deal with quadratic or cubic.
+        if len(control_points) == 3:
+            p0, p1, p2 = control_points
+            p0 = p0[i]
+            p1 = p1[i]
+            p2 = p2[i]
```

Comment:
This logic can be simplified by doing `control_points = np.array(control_points)` at the start of the function and then doing `p0, p1, p2 = control_points[:, i]`. Though I'd probably prefer doing `control_coords = control_points[:, i]` (or instead of `i`, `dim` as mentioned earlier) and then accessing them with `control_coords[0]` etc. instead of having a bunch of loose variables.